dnf-json

```diff
@@ -116,6 +116,52 @@ class Solver():
             })
         return packages
 
+    def search(self, args):
+        """ Perform a search on the available packages
+
+        TODO: Document parameters
+        "search" dict in transactions
+        "packages" list of strings in the search dict
+
+        """
+        pkg_globs = args.get("packages", [])
+
+        packages = []
+
+        # NOTE: Build query one piece at a time, don't pass all to filterm at the same
+        # time.
+        available = self.base.sack.query().available()
+        for name in pkg_globs:
+            # If the package name glob has * in it, use glob.
+            # If it has *name* use substr
+            # If it has neither use exact match
```

Comment:
What was the reason for this logic? Is it the same as dnf uses on the command line?

Reply:
No, this matches the behavior of the API queries (which IIRC were first written to use dnf via lorax-composer). I split it between globbing all the time and glob/substr hoping that there would be some speed difference.